apps/src/aichat/types.ts

```diff
@@ -17,6 +17,9 @@ export interface ChatEvent {
   // This allows the chat event to be visible by default without having to add an extra field.
   hideForParticipants?: true;
   descriptionKey?: keyof typeof ChatEventDescriptions;
+  // This field is optional but when it is defined, it must be set to `true`.
+  // This allows a particulat chat event to be visible only when in teacher view of student chat history.
+  isTeacherView?: true;
```

Comment:
I was thinking that we need a `isTeacherView` property so that we can allow teachers to toggle between hide/view student messages that are tagged as profane. But will hold off on adding until follow-up and will see if I can implement via props vs adding another property.